### PR TITLE
make eni return the consistent results as cni

### PR DESCRIPTION
**Pull request by richardpen**
Address issue #21 
make unit-test
make integration-test
Manually tested, no error will printed.

### CODE REVIEW

File: plugins/eni/commands/commands.go [hunk 1]
```diff
@@ -110,7 +113,27 @@ func add(args *skel.CmdArgs, engine engine.Engine) error {
 	}
 	log.Infof("ENI %s has been assigned to the container's namespace", conf.MACAddress)
 
-	return nil
+	_, ipv4Net, err := net.ParseCIDR(ipv4Gateway + "/" + ipv4Netmask)
```
Comment (vsiddharth) [inline plugins/eni/commands/commands.go]: I think we are better off using `Sprintf` here

File: plugins/eni/commands/commands.go [hunk 2]
```diff
@@ -110,7 +113,27 @@ func add(args *skel.CmdArgs, engine engine.Engine) error {
...
+				Name: networkDeviceName,
+				Mac:  macAddressOfENI,
+			},
+		},
+		IPs: []*current.IPConfig{
+			{
+				Version: "4",
+				Address: *ipv4Net,
+			},
+		},
```
Comment (vsiddharth) [inline plugins/eni/commands/commands.go]: Should we also stick the ipv6 address here ?

File: plugins/eni/commands/commands.go [hunk 3]
```diff
@@ -78,6 +81,17 @@ func add(args *skel.CmdArgs, engine engine.Engine) error {
 	}
 	log.Debugf("Found ipv4 gateway and netmask for ENI: %s %s", ipv4Gateway, ipv4Netmask)
 
+	_, ipv4Net, err := net.ParseCIDR(fmt.Sprintf("%s/%s", ipv4Gateway, ipv4Netmask))
+	if err != nil {
+		return errors.Wrapf(err, "add eni: parse cidr failed: %s", fmt.Sprintf("%s/%s", ipv4Gateway, ipv4Netmask))
```
Comment (aaithal) [inline plugins/eni/commands/commands.go]: Can you please add more context to this error message around where it's generated (or make it more customer friendly)?

`"add eni: failed to parse gateway netmask: %s"`

File: plugins/eni/commands/commands.go [hunk 4]
```diff
@@ -97,6 +111,15 @@ func add(args *skel.CmdArgs, engine engine.Engine) error {
 			return err
 		}
 		log.Debugf("IPV6 Gateway IP: %v", ipv6Gateway)
+		_, ipv6net, err := net.ParseCIDR(ipv6Address)
+		if err != nil {
+			return errors.Wrapf(err, "parsing ipv6 failed:%s", ipv6Address)
```
Comment (aaithal) [inline plugins/eni/commands/commands.go]: Same here, please make provide more context here: 

`"add eni: failed to parse ipv6 gateway: %s"`

File: plugins/eni/commands/commands.go [hunk 5]
```diff
@@ -15,11 +15,14 @@ package commands
...
 import (
 	"fmt"
+	"net"
 
 	"github.com/aws/amazon-ecs-cni-plugins/plugins/eni/engine"
 	"github.com/aws/amazon-ecs-cni-plugins/plugins/eni/types"
 	log "github.com/cihub/seelog"
 	"github.com/containernetworking/cni/pkg/skel"
+	cnitypes "github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
```
Comment (vsiddharth) [inline plugins/eni/commands/commands.go]: Nit: We should also separate out the 3p imports by adding a newline

File: plugins/eni/commands/commands.go [hunk 6]
```diff
@@ -78,6 +81,17 @@ func add(args *skel.CmdArgs, engine engine.Engine) error {
 	}
 	log.Debugf("Found ipv4 gateway and netmask for ENI: %s %s", ipv4Gateway, ipv4Netmask)
 
+	_, ipv4Net, err := net.ParseCIDR(fmt.Sprintf("%s/%s", ipv4Gateway, ipv4Netmask))
```
Comment (vsiddharth) [inline plugins/eni/commands/commands.go]: Nit: store `fmt.Sprintf("%s/%s", ipv4Gateway, ipv4Netmask)` as a string and make both ipv4 and ipv6 checks semantically consistent

File: plugins/eni/commands/commands_test.go [hunk 7]
```diff
@@ -15,12 +15,18 @@ package commands
 
 import (
 	"errors"
+	"io/ioutil"
+	"os"
 	"testing"
 
 	"github.com/aws/amazon-ecs-cni-plugins/plugins/eni/engine/mocks"
```
Comment (jhaynes) [inline plugins/eni/commands/commands_test.go]: Similar to @vsiddharth's Nit above: this should be in its own local import "chunk" shouldn't it?

**Pull request merged**